collector/iscsi_linux.go

```diff
@@ -0,0 +1,402 @@
+			"iSCSI FileIO backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "fileio", "object", "filename"}, nil,
+		),
+		lioFileWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioFileioSubsystem, "write_total"),
+			"iSCSI FileIO backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "fileio", "object", "filename"}, nil,
+		),
+
+		lioBlockIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioIblockSubsystem, "iops_total"),
+			"iSCSI IBlock backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "iblock", "object", "blockname"}, nil,
+		),
+		lioBlockRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioIblockSubsystem, "read_total"),
+			"iSCSI IBlock backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "iblock", "object", "blockname"}, nil,
+		),
+		lioBlockWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioIblockSubsystem, "write_total"),
+			"iSCSI IBlock backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "iblock", "object", "blockname"}, nil,
+		),
+
+		lioRbdIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "iops_total"),
+			"iSCSI RBD backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+		lioRbdRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "read_total"),
+			"iSCSI RBD backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+		lioRbdWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "write_total"),
+			"iSCSI RBD backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+
+		lioRdmcpIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "iops_total"),
+			"iSCSI Memory Copy RAMDisk backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "read_total"),
+			"iSCSI Memory Copy RAMDisk backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "write_total"),
+			"iSCSI Memory Copy RAMDisk backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+	}, nil
+}
+
+// /sys/kernel/config/target/iscsi/iqn*/tpgt_*/lun/lun_*/ which link
+// back to the following
+// /sys/kernel/config/target/core/{backstoreType}_{number}/{objectName}/
+func (c *lioCollector) updateStat(ch chan<- prometheus.Metric, s *iscsi.Stats) error {
+
+	log.Debugf("lio updateStat iscsi %s path", s.Name)
+	tpgtS := s.Tpgt
+	for _, tpgt := range tpgtS {
+		tpgtPath := tpgt.TpgtPath
+		iscsiEnable := tpgt.IsEnable
+
+		log.Debugf("lio: iscsi %s isEnable=%t", tpgtPath, iscsiEnable)
```

Comment:
`tpgt.TpgtPath` and `tpgt.IsEnable` can be used directly here so that the local variables are no longer needed.

Reply:
Thanks removed and push